src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -644,6 +644,88 @@ protected override void ApplyAnalyzerReferenceRemoved(ProjectId projectId, Analy
             }
         }
 
+        protected override void ApplyProjectChanges(ProjectChanges projectChanges)
+        {
+            base.ApplyProjectChanges(projectChanges);
```

Comment:
i don't really understand this.  if we call into the base, won't that try to to apply text changes as well?  needs docs.

Reply:
Yep, it will apply text changes to documents in the workspace, but not mapped text changes (changes in razor files).  Will update the docs.

Reply:
what prevents that?

Reply:
The razor documents have CanApplyChange == false, so the regular workspace document changes application skips them.